### DISCUSSION TITLE
Custom jsonUrl host for Docs handler

### DISCUSSION THREAD

**Discussion by wangha81** [meteran/gnext#21] [Ideas]
I found `jsonUrl` was hardcode by "http://localhost:%s%s" only for localhost

Hope it can setup by args pass by default localhost like "http://<host>:%s%s"

The code was at [docs/handler.go](https://github.com/meteran/gnext/blob/6eedd93653b5670dde5aa2f5dfc9b4940dea14af/docs/handler.go#L32)

**Comment by meteran** [comment]
Hi @wangha81, thanks for your interest in gnext.

This docs page is just for testing/debugging purpose. For the production environment you should use some self hosted documentation page. You can use `SaveAsJson` or `SaveAsYaml` methods on `Docs` struct to generate the openapi docs file and serve them using a CDN.

However, for smaller projects it might be indeed helpful to define the custom public URL. So I will add such option in next version. I will let you know when it's implemented.

**Comment by meteran** [comment]
I've changed the URL to the relative one (just `JsonUrl` value). It will work till both - docs and swagger UI are at the same locations(host:port or domain). I created a pull request https://github.com/meteran/gnext/pull/23 . You can try it from my branch. I will release it tomorrow.

Please also note, that it won't work if files are in different locations. To cover such a case I would have to add another docs option. But I don't think this is a good idea. If someone hosts those files in different locations, they are probably in CDN or other file server (better and recommended way as I mentioned in a previous post).

**Comment by meteran** [comment]
I've released the version v0.8.0 with relative docs path. Does it work for you? If so, I will close the discussion.

**Reply by wangha81** [reply]
Thanks a lot, I'll update my little walk around version. It's helpful.
